### PR TITLE
Package fields

### LINKED ISSUE #416: Restore `main`, `module`, and `types`
The `exports` field is still fairly new for Typescript users, most of whom still have `"moduleResolution": "node"` in their `tsconfig.json`s. Such is the case of this repo as well.

The latest PR #404 from @benmccann removed the `main`, `module`, and `types` fields which node relies on for its module resolution.

```
src/blur/index.ts(1,29): error TS2307: Cannot find module 'imagetools-core' or its corresponding type declarations.
src/blur/index.ts(4,39): error TS2307: Cannot find module 'imagetools-core' or its corresponding type declarations.
src/blur/index.ts(42,33): error TS7006: Parameter 'config' implicitly has an 'any' type.
error Command failed with exit code 1.

```

AFAIK, the aforementioned fields can coexist with the `exports` fields.

### PR DESCRIPTION
**Pull request by scheatkode**
- **Quick Checklist**

* [x] I have read [the contributing guidelines](../CONTRIBUTING.md)
* [ ] I have written new tests, as applicable (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [x] I have added a changeset, if applicable

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes #416.

- **What is the new behavior (if this is a feature change)?**

There should be better compatibility with node module resolution while still keeping the `exports` field.

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this
  PR?)

No breaking change, simply a hotfix.

- **Other information**:

### CODE REVIEW

**Comment by benmccann** [comment]
Do we need all of these or just the `types` field? I'm not as familiar with TypeScript, but Node, Vite, etc. will ignore `main` and `module` now that `exports` is present

**Comment by scheatkode** [comment]
You're right, I just made a quick test in some unrelated repository and it did work.

I banged my head at this so much in the past I didn't bother anymore. Thanks for the clarification !

**Comment by JonasKruckenberg** [comment]
Nice hahaha you guys beat me to it 😄

File: .changeset/tiny-dryers-sneeze.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'imagetools-core': patch
+'rollup-plugin-imagetools': patch
+'vite-imagetools': patch
+---
+
+fix: restore package.json fields
```
Comment (JonasKruckenberg) [inline .changeset/tiny-dryers-sneeze.md:7]:
```suggestion
Fix type resolution by re-adding top-level types field.
```

**Review by JonasKruckenberg — COMMENTED** [review]
Just reword the changeful now to reflect the updates and were good

**Comment by scheatkode** [comment]
All done ! Apologies for the late reply and thank you both for your time and work

**Pull request merged**